modules/core/test/test_intrin_utils.hpp

```diff
@@ -1794,6 +1794,76 @@ template<typename R> struct TheTest
         return *this;
     }
 
+    TheTest & test_exp_fp32()
+    {
+        int n = VTraits<R>::vlanes();
```

Comment:
The test is very weak. v_exp() is critical function. We must test it properly.

1. We must test all the corner and popular cases

```
exp(nan)=nan,
exp(0)=1,
exp(1)~e,
exp(inf)=inf,
exp(-inf)=0,
exp(88) < inf,
exp(89..100)=inf (overflow), // check every number with step 0.1 or even 0.01
0<=exp(-100..-88)<FLT_MIN (underflow). // check every number with step 0.1 or even 0.01
```

2.

```
for (int i = 0; i < 10000; i++) {
    1. generate 4 numbers x_j, j=0..3 in range [-100,100].
    2. compute vector exponent v_exp()
    3. compute exponent for each number using std::exp()
    4. check relative accuracy
         a. std::exp(x_j) = inf => EXPECT_GT(vector_exp_j, FLT_MAX*0.99)
         b. std::exp(x_j) < inf =>
                    EXPECT_GE(vector_exp_j, 0);
                    EXPECT_LT(std::abs(vector_exp_j-std_exp_j) < 1e-6*(std::abs(std_exp_j)+FLT_MIN*100)
```

where 0.99, 1e-6 and 100 should probably be adjusted.

Reply:
Thanks, I will add tests for these corner cases.

Reply:
since `v_max(nan, x)` and `v_min(nan, x)` will not return `nan` except neon, I disabled this test. If there are some way to handle this cases?